gateways/rocketgate/rocketgate.go

```diff
@@ -4,42 +4,126 @@ import (
 	"net/http"
 
 	"github.com/BoltApp/sleet"
+	"github.com/BoltApp/sleet/common"
+	"github.com/BoltApp/sleet/rocketgate-sdk/response"
+	"github.com/BoltApp/sleet/rocketgate-sdk/service"
 )
 
 // RocketgateClient represents an HTTP client and the associated authentication information required for
 // making an API request.
 type RocketgateClient struct {
-	httpClient      *http.Client
+	testMode 			bool
+	merchantID  		string
+	merchantPassword	string
+	merchantAccount 	*string
+	httpClient  		*http.Client
 }
 
 // NewClient creates a Rocketgate client
-func NewClient(httpClient *http.Client) *RocketgateClient {
-	return NewWithHttpClient(httpClient)
+func NewClient(env common.Environment, merchantID string, merchantPassword string, merchantAccount *string, httpClient *http.Client) *RocketgateClient {
+	return NewWithHttpClient(env, merchantID, merchantPassword, merchantAccount, httpClient)
 }
 
 // NewWithHttpClient creates an Rocketgate client for custom behavior
-func NewWithHttpClient(httpClient *http.Client) *RocketgateClient {
+func NewWithHttpClient(env common.Environment, merchantID string, merchantPassword string, merchantAccount *string, httpClient *http.Client) *RocketgateClient {
```

Comment:
Sorry, so I thought we want this func still pass `httpClient *http.Client` instead of `DefaultHTTPClient` so user still can pass their custom httpClient? (and this is what other gateway do? but please correct me if I am wrong)

Reply:
oh, or you just mean "let's put each param on its own line, this line is a bit long as is." ?

Reply:
Sorry this comment was confusing, just meant the "each param on its own line" part.